doc/kernel/usermode/memory_domain.rst

```diff
@@ -74,10 +74,24 @@ Thread Stack
 ************
 
 Any thread running in user mode will need access to its own stack buffer.
-On context switch into a user mode thread, a dedicated MPU region will be
-programmed with the bounds of the stack buffer. A thread exceeding its stack
-buffer will start pushing data onto memory it doesn't have access to and a
-memory access violation exception will be generated.
+On context switch into a user mode thread, a dedicated MPU region or MMU
+page table entries will be programmed with the bounds of the stack buffer.
+A thread exceeding its stack buffer will start pushing data onto memory
+it doesn't have access to and a memory access violation exception will be
+generated.
+
+Note that threads have access to the stacks of other threads in the same
+memory domain. This is the default configuration and the minimum
+required to support memory domain. Architecture can further restrict
+access to stacks so each thread only has access to its own stack if
+such architecture advertises this capability via
+:kconfig:option:`CONFIG_ARCH_MEM_DOMAIN_SUPPORTS_ISOLATED_STACKS`.
+This behavior can be selectively enabled via
+:kconfig:option:`CONFIG_MEM_DOMAIN_ISOLATED_STACKS` if architecture
```

Comment:
Shouldn't this be the default behavior? It's actually a bit surprising to see that we would need to opt into the stricter behavior

Reply:
Default is to enable the feature if supported. I was trying to convey the idea that this could be toggled. Changed the wording to clarify that being enabled is the default.